src/lib/core/TLVWriter.h

```diff
@@ -1181,26 +1201,27 @@ class DLL_EXPORT TLVWriter
      * @note The value of the @p ImplicitProfileId member affects the encoding of profile-specific
      * tags only; the encoding of context-specific tags is unchanged.
      */
-    uint32_t ImplicitProfileId;
+    uint32_t ImplicitProfileId = kProfileIdNotSpecified;
 
     /**
      * A pointer field that can be used for application-specific data.
      */
-    void * AppData;
+    void * AppData = nullptr;
 
 protected:
-    TLVBackingStore * mBackingStore;
-    uint8_t * mBufStart;
-    uint8_t * mWritePoint;
-    uint32_t mRemainingLen;
-    uint32_t mLenWritten;
-    uint32_t mMaxLen;
-    uint32_t mReservedSize;
-    TLVType mContainerType;
+    TLVBackingStore * mBackingStore = nullptr;
+    uint8_t * mBufStart             = nullptr;
```

Comment:
This seems to cause a non-negligible code size increase. Would making sure that all fields are initialized in `Init()` method silence that static code analyzer?

Reply:
all is 0 except kTLVType_NotSpecified which is -1 and mCloseContainerReserved which is false. I wonder if that moves things out of 'initialize with 0' into more complex code.

Reply:
Changed the code to only be calling the default constructor. If it grows by 3k+, where are all the TLVWriters? Single callsite of `TLVWriter x;` default constructor is <= 16 bytes total.